src/backend/lib/upgrade-manager.js

```diff
@@ -0,0 +1,147 @@
+const UpgradeStorage = require("./upgrade-storage");
+const UpgradeServer = require("./upgrade-server");
+const UpgradeDownload = require("./upgrade-download");
+const semver = require("semver");
+const log = require("debug")("p2p-upgrades:manager");
+const validate = require("./validate-upgrade-state");
+
+class UpgradeManager {
+  constructor(dir, port, currentVersion, emitFn, listenFn, removeFn) {
```

Comment:
```suggestion
  constructor({ dir, port, currentVersion, emitFn, listenFn, removeFn}) {
```

Code style pref: With this many arguments it is easy for a mistake to creep in when calling the constructor by missing an argument or getting them in the wrong order (I've spent some time debugging issues like this). Passing arguments as properties on an `options` object has less chance of error, and is easy with destructuring like this.